app/screens/invite/invite.tsx

```diff
@@ -0,0 +1,419 @@
+
+export type InviteResult = {
+    userId: string;
+    reason: string;
+};
+
+export type Result = {
+    sent: InviteResult[];
+    notSent: InviteResult[];
+}
+
+enum Stage {
+    SELECTION = 'selection',
+    RESULT = 'result',
+    LOADING = 'loading',
+}
+
+type InviteProps = {
+    componentId: string;
+    teamId: string;
+    teamDisplayName: string;
+    teamLastIconUpdate: number;
+    teamInviteId: string;
+    teammateNameDisplay: string;
+    isAdmin: boolean;
+}
+
+export default function Invite({
+    componentId,
+    teamId,
+    teamDisplayName,
+    teamLastIconUpdate,
+    teamInviteId,
+    teammateNameDisplay,
+    isAdmin,
+}: InviteProps) {
+    const intl = useIntl();
+    const {formatMessage, locale} = intl;
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+    const mainView = useRef<View>(null);
+    const modalPosition = useModalPosition(mainView);
+
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const retryTimeoutId = useRef<NodeJS.Timeout | null>(null);
+
+    const [term, setTerm] = useState('');
+    const [searchResults, setSearchResults] = useState<SearchResult[]>([]);
+    const [selectedIds, setSelectedIds] = useState<{[id: string]: SearchResult}>({});
+    const [loading, setLoading] = useState(false);
+    const [result, setResult] = useState<Result>(DEFAULT_RESULT);
+    const [wrapperHeight, setWrapperHeight] = useState(0);
+    const [stage, setStage] = useState(Stage.SELECTION);
+    const [sendError, setSendError] = useState('');
+
+    const selectedCount = Object.keys(selectedIds).length;
+
+    const onLayoutWrapper = useCallback((e: LayoutChangeEvent) => {
+        setWrapperHeight(e.nativeEvent.layout.height);
+    }, []);
+
+    const searchUsers = useCallback(async (searchTerm: string) => {
+        if (searchTerm === '') {
+            handleClearSearch();
+            return;
+        }
+
+        const {data} = await searchProfiles(serverUrl, searchTerm.toLowerCase());
+        const results: SearchResult[] = data ?? [];
+
+        if (!results.length && isEmail(searchTerm.trim())) {
+            results.push(searchTerm.trim() as EmailInvite);
+        }
+
+        setSearchResults(results);
+    }, [serverUrl, teamId]);
+
+    const handleReset = () => {
+        setStage(Stage.LOADING);
+        setSendError('');
+        setTerm('');
+        setSearchResults([]);
+        setResult(DEFAULT_RESULT);
+        setStage(Stage.SELECTION);
+    };
+
+    const handleClearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults([]);
+    }, []);
+
+    const handleSearchChange = useCallback((text: string) => {
+        setLoading(true);
+        setTerm(text);
+
+        if (searchTimeoutId.current) {
+            clearTimeout(searchTimeoutId.current);
+        }
+
+        searchTimeoutId.current = setTimeout(async () => {
+            await searchUsers(text);
```

Comment:
@larkox this is a good example on why we would need to cancel a previous request.